src/components/inputs/rich-text-input/dropdown.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+import Tooltip from '../../tooltip';
+import AccessibleButton from '../../buttons/accessible-button';
+import {
+  getButtonStyles,
+  DropdownItem as StyledDropdownItem,
+  DropdownContainer,
+} from './dropdown.styles';
+
+// eslint-disable-next-line
+const Label = props => <span>{props.children}</span>;
+
+const Dropdown = props => {
+  const DropdownItem = props.components.DropdownItem || StyledDropdownItem;
+  const DropdownLabel = props.components.DropdownLabel || Label;
+  return (
+    <Downshift
+      onChange={props.onChange}
+      selectedItem={props.value}
+      itemToString={headings => (headings ? headings.label : '')}
+    >
+      {({
+        isOpen,
+        toggleMenu,
+        getToggleButtonProps,
+        getItemProps,
+        highlightedIndex,
+        selectedItem,
+      }) => {
+        const { onClick, ...restOfToggleButtonProps } = getToggleButtonProps();
+        return (
+          <div>
+            <Tooltip
+              title="Text styles"
+              placement="bottom"
+              off={isOpen}
+              style={{ body: { zIndex: 9999 } }}
+            >
+              <AccessibleButton
+                label={props.label}
+                css={getButtonStyles()}
+                {...restOfToggleButtonProps}
+                onMouseDown={toggleMenu}
+              >
+                <DropdownLabel>{props.label}</DropdownLabel>
+              </AccessibleButton>
+            </Tooltip>
+            <div style={{ position: 'relative' }}>
+              {isOpen ? (
+                <DropdownContainer>
+                  {props.options.map((item, index) => {
+                    const {
+                      onClick: dropdownOnClick,
+                      ...restOfItemProps
+                    } = getItemProps({
```

Comment:
Why do we destructure here? Can't we below to `itemProps.onClick` and if we don't want to pass on `onClick` to it explicitely with an `omit(itemProps, ['onClick'])`.

Reply:
yeah can use omit